code/modules/power/apc.dm

```diff
@@ -699,20 +708,69 @@
 			)
 		)
 	)
+	return data
 
-	//Update the ui if it exists, returns null if no ui is passed/found
-	ui = SSnano.try_update_ui(user, src, ui_key, ui, data, force_open)
-	if (!ui)
-		//The ui does not exist, so we'll create a new() one
-		//For a list of parameters and their descriptions see the code docs in \code\modules\nano\nanoui.dm
-		ui = new(user, src, ui_key, "apc.tmpl", "[area.name] - APC", 520, data["siliconUser"] ? 465 : 440)
-		//When the ui is first opened this is the data it will use
-		ui.set_initial_data(data)
-		//Open the new ui window
-		ui.open()
-		//Auto update every Master Controller tick
-		ui.set_auto_update(1)
 
+/obj/machinery/power/apc/proc/setsubsystem(val)
+	if(cell && cell.charge > 0)
+		return (val==1) ? 0 : val
+	else if(val == 3)
+		return 1
+	else
+		return 0
+
+/obj/machinery/power/apc/proc/can_use(mob/user, loud = 0) //used by attack_hand() and Topic()
+	if(IsAdminGhost(user))
+		return TRUE
+	if(isAI(user) && src.aidisabled)
+		if(!loud)
+			to_chat(user, "<span class='danger'>\The [src] has eee disabled!</span>")
+		return FALSE
+	return TRUE
+
+/obj/machinery/power/apc/ui_act(action, params)
+	if(..() || !can_use(usr, 1) || locked)
+		return
+	switch(action)
+		if("lock")
+			if(usr.has_unlimited_silicon_privilege)
+				if((machine_stat & (BROKEN|MAINT)))
+					to_chat(usr, "The APC does not respond to the command.")
+				else
+					locked = !locked
+					update_icon()
+					. = TRUE
+		if("cover")
+			coverlocked = !coverlocked
+			. = TRUE
+		if("breaker")
+			toggle_breaker(usr)
+			. = TRUE
+		if("charge")
+			chargemode = !chargemode
+			if(!chargemode)
+				charging = APC_NOT_CHARGING
+				update_icon()
+			. = TRUE
+		if("channel")
+			if(params["eqp"])
+				equipment = setsubsystem(text2num(params["eqp"]))
+				update_icon()
+				update()
+			else if(params["lgt"])
+				lighting = setsubsystem(text2num(params["lgt"]))
+				update_icon()
+				update()
+			else if(params["env"])
+				environ = setsubsystem(text2num(params["env"]))
+				update_icon()
+				update()
+			. = TRUE
+		if("overload")
+			if(usr.has_unlimited_silicon_privilege)
+				overload_lighting()
+				. = TRUE
+	return 1
```

Comment:
Why all these checks just to `return 1`?

Reply:
ask tg

Reply:
Please don't just resolve things like this.
The return seems to determine whether the UI should update or not, so this is safe to remove.